yt_dlp/extractor/zingmp3.py

```diff
@@ -51,45 +83,88 @@ def _extract_item(self, item, fatal):
         self._sort_formats(formats)
 
-        subtitles = None
         lyric = item.get('lyric')
-        if lyric:
-            subtitles = {
-                'origin': [{
-                    'url': lyric,
-                }],
-            }
+        if not lyric:
+            api = self.get_api_with_signature(name_api=SLUG_API.get("lyric"), param={
+                'ctime': TIMESTAMP,
+                'id': item_id})
+            info_lyric = self.download_json(api, item_id)
+            lyric = traverse_obj(info_lyric, ('data', 'file'))
+        subtitles = {
+            'origin': [{
+                'url': lyric,
+            }],
+        } if lyric else None
 
         album = item.get('album') or {}
 
         return {
             'id': item_id,
             'title': title,
             'formats': formats,
-            'thumbnail': item.get('thumbnail'),
+            'thumbnail': item.get('thumbnail') or item.get('thumbnailM'),
             'subtitles': subtitles,
             'duration': int_or_none(item.get('duration')),
             'track': title,
-            'artist': item.get('artists_names'),
+            'artist': item.get('artistsNames') or item.get('artists_names'),
             'album': album.get('name') or album.get('title'),
-            'album_artist': album.get('artists_names'),
+            'album_artist': album.get('artistsNames') or album.get('artists_names'),
         }
 
     def _real_extract(self, url):
-        page_id = self._match_id(url)
-        webpage = self._download_webpage(
-            url.replace('://zingmp3.vn/', '://mp3.zing.vn/'),
-            page_id, query={'play_song': 1})
-        data_path = self._search_regex(
-            r'data-xml="([^"]+)', webpage, 'data path')
-        return self._process_data(self._download_json(
-            'https://mp3.zing.vn/xhr' + data_path, page_id)['data'])
+        song_id, type_url = self._match_valid_url(url).group('id', 'type')
+
+        slug_api = SLUG_API.get(type_url)
+
+        # Create api from slug_api with hash signature
+        api = self.get_api_with_signature(name_api=slug_api, param={
+            'ctime': TIMESTAMP,
+            'id': song_id})
+        info = self.download_json(api, song_id)
+        return self._process_data(info.get('data'), song_id, type_url)
+
+    def download_json(self, api, video_id):
+        # For the first time making the request, we need to make one more temp request to update the cookie
+        # and the next time we don't need to do it.
+        # But if download with cookieFile, so we don't need to make one more temp request
+        if self._downloader.params.get('cookiefile'):
+            self._IS_UPDATE_COOKIES = True
+
+        if not self._IS_UPDATE_COOKIES:
+            self._download_json(api, video_id=video_id, note='Update cookies for request')
+            self._IS_UPDATE_COOKIES = True
+
+        return self._download_json(api, video_id=video_id)
```

Comment:
I honestly don't understand why this function is required tbh.
Why can't u just manually make a call once?
Why is this func required  if it's going to behave like the builtin one after being called for the first time?

Reply:
Actually, i tried to make a call once, but it's allway error without cookie which received from the first call.

Reply:
And make one more call for the first time, it will not affect to the performance, i think it's okay now.

Reply:
this us not resolved!

Reply:
As i explained above, adding 1 garbage request only do it once, and next time won't need it, if you think it's not good, do you have a better solution for me?

Reply:
I should have worded my original comment better.
I didn't mean to say that make the actual call once.
What i meant was,

That from my understanding of this code, the API requires a cookie to be set before it starts sending u actual data.

So how this works rn (As i understand it)
1. u make a req at endpoint 1
2. it makes an if check for the cookie flag
3. it makes dummy req at endpoint 1
4. it sets that cookie flag to be true
5. it makes actual data request at endpoint 1
6. u make req at endpoint 2
7. it makes an if check for that cookie flag
8. it makes actual data request at endpoint 2
and so on...

What I'm suggesting is, eliminate this cookie flag check and do

1. make a req for the cookie
2. make actual data req for endpoint1.
3. make actual data req for endpoint2.
and so on

Reply:
It's been a while since i looked at this code so don't rember the exact endpoint names.
If it works how I am suggesting, then u won't need ur own download_json function.

we do the process of making a dummy request just for cookie file in other extractors as well.

Reply:
I understood, i will check if cookie flag doesnt exist, i will create a dummy req to update the cookie
The we can make a req to endpoint1
And endpoint2 and so on...